### PR TITLE
free all entities after fini

### PR DESCRIPTION
**Pull request by dirk-thomas**
Same as #109 but for all other entities.

### CODE REVIEW

File: rclpy/src/rclpy/_rclpy.c [hunk 1]
```diff
@@ -1356,19 +1356,31 @@ rclpy_destroy_node_entity(PyObject * Py_UNUSED(self), PyObject * args)
     Py_RETURN_FALSE;
   }
 
-  rcl_node_t * node = (rcl_node_t *)PyCapsule_GetPointer(pynode, NULL);
+  void * p = PyCapsule_GetPointer(pynode, NULL);
+  if (Py_None == p) {
```
Comment (allenh1) [inline rclpy/src/rclpy/_rclpy.c]: @dirk-thomas Just wondering why this is `Py_None == p` as opposed to `p == Py_None`. Is this style, or is this something deeper?

Comment (dirk-thomas) [reply rclpy/src/rclpy/_rclpy.c]: This style aims to put the constant on the left to avoid accidental assignments with `=` instead of `==`.

Comment (allenh1) [reply rclpy/src/rclpy/_rclpy.c]: That makes a lot of sense! Thank you!

**Comment by dirk-thomas** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/3162/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/500/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/2519/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/3213/)

**Pull request merged**